src/data/reducers/index.js

```diff
@@ -3,9 +3,14 @@ import { combineReducers } from 'redux';
 import assets from './assets';
 import connectionStatus from './connectionStatus';
 
+/* eslint-disable no-undef */
+/* eslint-disable arrow-body-style */
+const courseDetails = () => { return courseContext || {}; };
```

Comment:
Should we make the default here something more than an empty object? For instance, defaults for all of the different values that would be passed on?

Reply:
Maybe not even defaulting? You could throw an error if not set?

Reply:
Maybe not even defaulting? You could throw an error if not set?

Reply:
I added the || empty object due to an error I was seeing with redux when the application was starting up because of the `dispatch` in AssetsPage setup:

`dispatch => ({
    getAssets: (assetsParameters, courseDetails) =>
      dispatch(getAssets(assetsParameters, courseDetails)),
  }),`

Let me test again to validate if that error is still occurring if I remove the default.

Reply:
@dsjen - removing the default no longer errors out :) Must have been a byproduct of previous logic. Is there any semi-standard way edX handles throwing/catching errors in javascript?

Reply:
You are paving ways! I think we can throw something like, "Course details required" or something like that.